packages/repocop/src/rules/repository.ts

```diff
@@ -422,6 +427,39 @@ export function testExperimentalRepocopFeatures(
 	);
 }
 
+export function deduplicateVulnerabilities(
+	vulns: RepocopVulnerability[],
+): RepocopVulnerability[] {
+	const vulnsWithSortedCVEs = vulns.map((v) => {
+		return {
+			...v,
+			CVEs: v.CVEs.sort(),
+		};
+	});
+	const [withCVEs, withoutCVEs] = partition(
+		vulnsWithSortedCVEs,
+		(v) => v.CVEs.length > 0,
+	);
+
+	const criticalFirstPredicate = (x: RepocopVulnerability) =>
+		x.severity === 'critical' ? -1 : 1;
+
+	//group withCVEs by CVEs
+	const dedupedWithCVEs = withCVEs
+		.sort(criticalFirstPredicate)
+		.reduce<Record<string, RepocopVulnerability>>((acc, vuln) => {
+			const key = vuln.CVEs.join(',');
+			if (!acc[key]) {
+				acc[key] = vuln;
+			}
+			return acc;
+		}, {});
+
+	const x: RepocopVulnerability[] = Object.values(dedupedWithCVEs);
+	const dedupedVulns = x.concat(withoutCVEs);
```

Comment:
nit: could we inline `x` here?

```suggestion
	const dedupedVulns = Object.values(dedupedWithCVEs).concat(withoutCVEs);
```

Reply:
whoops!